tests/kafkatest/services/console_consumer.py

```diff
@@ -102,6 +100,10 @@ def __init__(self, context, num_nodes, kafka, topic, group_id="test-consumer-gro
         for node in self.nodes:
             node.version = version
 
+        self.jmx_tool = JmxTool(context, num_nodes=num_nodes,
+                                jmx_object_names=jmx_object_names, jmx_attributes=(jmx_attributes or []),
+                                root=ConsoleConsumer.PERSISTENT_ROOT)
```

Comment:
@mumrah to your earlier point, I've tried to avoid special casing by always initializing a `JmxTool` regardless of whether it'll be used

@hachikuji however, now that this is in place, it seems like we should maybe just stick with mixin approach since we're always initializing this object anyways? (this way we can also avoid some of the `clean_node` changes required here)

with that being said, I do like the `consumer.jmx_tool` namespace isolation

Reply:
@brianbushree I don't mind going back to the mixin if you think that's better. I'll leave it up to you.

Reply:
I think this mixin approach is better for two reasons here:
1. it should minimize this PR
2. seems less like a one-off approach since we already use the mixin everywhere

I'm going to refactor this PR accordingly.